### PR TITLE
fix(sql): precision is lost when displaying timestamps

### LINKED ISSUE #1671: [Bug]: precision is lost when displaying timestamps
### ODC version

4.2.4

### OB version

not related

### What happened?

precision is lost when displaying timestamps

### What did you expect to happen?

precision should not be lost

### How can we reproduce it (as minimally and precisely as possible)?

1. create a table and insert table:
```sql
create table ts_tbl(
    col_ts_3 timestamp(3),
    col_ts_6 timestamp(6)
);
insert into ts_tbl values('2022-12-12 12:12:12.0','2022-12-12 12:12:12.0');
```
2. query this data:
截屏2024-02-29 16 43 07

we will see more precision if we query this data by obclient:
截屏2024-02-29 16 44 11

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-bug

#### Which issue(s) this PR fixes:

Fixes #1671

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLTimestampMapper.java [hunk 1]
```diff
@@ -41,10 +41,11 @@ public Object mapCell(@NonNull CellData data) throws SQLException {
         if (!isValidTimestamp(data)) {
             return "0000-00-00 00:00:00";
         }
-        String returnValue = timestamp.toString();
-        if (returnValue.endsWith(".0")) {
-            return returnValue.substring(0, returnValue.length() - 2);
-        }
+
+        String returnValue = new String(data.getBytes());
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLTimestampMapper.java:45]: remove useless code, you may not put them in a comment~

and you can return it directly.

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLTimestampMapper.java [hunk 2]
```diff
@@ -41,10 +41,11 @@ public Object mapCell(@NonNull CellData data) throws SQLException {
...
+        /*
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLTimestampMapper.java:46]: btw, you may test this for two situation:
1. query data(you may have done this already)
2. try to modify a timestamp in table data viewing panel or result set viewing panel~

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by LuckyPickleZZ — APPROVED** [review]
lgtm

**Pull request merged**